core/src/main/java/org/infinispan/topology/ClusterCacheStatus.java

```diff
@@ -692,6 +694,17 @@ public void setRebalanceEnabled(boolean enabled) {
       }
    }
 
+   /**
+    * Only performs rebalance if the per-cache rebalancing is enabled
+    */
+   public void conditionalRebalance() {
```

Comment:
I think it's ok the way it is now, we want to broadcast the stable topology when a rebalance finishes, even if rebalancing is now disabled. We're doing it here because sometimes the stable topology can change without a rebalance (e.g. a node leaves in a replicated cache), and `startQueuedRebalance` happens to be called in all cases.